youtube_dl/extractor/ximalaya.py

```diff
@@ -0,0 +1,184 @@
+    _TEMPLATE_URL = 'http://www.ximalaya.com/%s/album/%s/'
+    _BASE_URL_TEMPL = 'http://www.ximalaya.com%s'
+    _TESTS = [{
+        'url': 'http://www.ximalaya.com/61425525/album/5534601/',
+        'info_dict': {
+            'title': 'contains:唐诗三百首（含赏析）',
+            'id': '5534601',
+        },
+        'playlist_count': 312,
+    }, {
+        'url': 'http://m.ximalaya.com/61425525/album/5534601',
+        'info_dict': {
+            'title': 'contains:唐诗三百首（含赏析）',
+            'id': '5534601',
+        },
+        'playlist_count': 312,
+    },
+    ]
+
+    def _real_extract(self, url):
+        uid, playlist_id = self._match_uid_an_id(url)
+        assert uid.isdecimal()
+        webpage = self._download_webpage(self._TEMPLATE_URL % (uid, playlist_id), playlist_id,
+                                         note='Download album page for %s' % playlist_id,
+                                         errnote='Unable to get album info'
+                                         )
+
+        mobj = re.search(r'detailContent_title(?:[^>]+)?><h1(?:[^>]+)?>([^<]+)</h1>', webpage)
+        title = mobj.group(1) if mobj else self._meta_regex('title')
+
+        return self.playlist_result(self._entries(webpage, playlist_id, uid), playlist_id, title)
+
+    def _entries(self, page, playlist_id, uid):
+        html = page
+        for page_num in itertools.count(1):
+            for entry in self._process_page(html, uid):
+                yield entry
+
+            mobj = re.search(r'<a href=(["\'])(?P<more>[^\'"]+)\1'
+                             r'[^>]+rel=(["\'])next\3', html)
+            if not mobj:
+                break
+
+            next_url = self._BASE_URL_TEMPL % mobj.group('more')
+            self.report_download_webpage('%d %s' % (page_num, next_url))
+            html = self._download_webpage(next_url, playlist_id)
+            if not html.strip():
+                # Some webpages show a "Load more" button but they don't
+                # have more videos
+                break
+
+    def _process_page(self, html, uid):
+        find_from = html.index('album_soundlist')
+        for mobj in re.finditer(r'<a[^>]+?href="(?P<url>/' +
+                                uid +
+                                r'/sound/(?P<id>\d+)/?)"[^>]+?title="(?P<title>[^>]+)">',
+                                html[find_from:]):
+            if 'url' in mobj.groupdict():
+                yield self.url_result(self._BASE_URL_TEMPL % mobj.group('url'),
+                                      'Ximalaya',
+                                      mobj.group('id'),
+                                      mobj.group('title'))
+
+    @classmethod
+    def _match_uid_an_id(cls, url):
+        if '_VALID_URL_RE' not in cls.__dict__:
+            cls._VALID_URL_RE = re.compile(cls._VALID_URL)
+        m = cls._VALID_URL_RE.match(url)
+        assert m
+        return compat_str(m.group('uid')), compat_str(m.group('id'))
```

Comment:
Use the mobj idiom instead. For example:
https://github.com/rg3/youtube-dl/blob/72c3d02d294b04b35a19417b31ad497e7540caa3/youtube_dl/extractor/teachingchannel.py#L29-L30

And is it necessary to create a function for this?